.github/workflows/ci-beta-nightly.yml

```diff
@@ -0,0 +1,58 @@
+name: Build Test (Beta and Nightly)
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-beta-nightly:
+    name: "Build for ${{ matrix.config.TARGET }} using Rust ${{ matrix.TOOLCHAIN }} (on ${{ matrix.config.OS }}) [args: ${{ matrix.BUILD_ARGS }}]"
```

Comment:
Maybe we can remove the "using Rust $TOOLCHAIN" here because we already have it in the workflow name.